### PR TITLE
Add bound 3PID warning when changing IS as well

### PR DESCRIPTION
**Pull request by jryans**
This extends the bound 3PID warning from the disconnect button to also appear
when changing the IS as well. At the moment, the text is a bit terse, but will
be improved separately.

2019-09-05 at 17 46

Fixes https://github.com/vector-im/riot-web/issues/10749

### CODE REVIEW

File: src/components/views/settings/SetIdServer.js [hunk 1]
```diff
@@ -157,14 +162,34 @@ export default class SetIdServer extends React.Component {
...
                 await authClient.getAccessToken();
 
+                let save = true;
+
                 // Double check that the identity server even has terms of service.
                 const terms = await MatrixClientPeg.get().getTerms(SERVICE_TYPES.IS, fullUrl);
                 if (!terms || !terms["policies"] || Object.keys(terms["policies"]).length <= 0) {
-                    this._showNoTermsWarning(fullUrl);
-                    return;
+                    save &= await this._showNoTermsWarning(fullUrl);
```
Comment (turt2live) [inline src/components/views/settings/SetIdServer.js]: fancy use of bitwise

File: src/components/views/settings/SetIdServer.js [hunk 2]
```diff
@@ -179,73 +204,80 @@ export default class SetIdServer extends React.Component {
...
-                    </span>
-                </div>
-            ),
-            button: _t("Continue"),
-            onFinished: async (confirmed) => {
-                if (!confirmed) return;
-                this._saveIdServer(fullUrl);
-            },
+        return new Promise(resolve => {
+            Modal.createTrackedDialog('No Terms Warning', '', QuestionDialog, {
```
Comment (turt2live) [inline src/components/views/settings/SetIdServer.js]: thanks to t3chguy this now returns access to a thenable - we should use that

File: src/components/views/settings/SetIdServer.js [hunk 3]
```diff
@@ -157,14 +162,34 @@ export default class SetIdServer extends React.Component {
...
                 if (!terms || !terms["policies"] || Object.keys(terms["policies"]).length <= 0) {
-                    this._showNoTermsWarning(fullUrl);
-                    return;
+                    save &= await this._showNoTermsWarning(fullUrl);
                 }
 
-                this._saveIdServer(fullUrl);
+                // Show a general warning, possibly with details about any bound
+                // 3PIDs that would be left behind.
+                if (this.state.currentClientIdServer) {
```
Comment (turt2live) [inline src/components/views/settings/SetIdServer.js]: `&& isDifferent`?

File: src/components/views/settings/SetIdServer.js (hunk 2 above)
Comment (jryans) [reply src/components/views/settings/SetIdServer.js]: Changed to this format, thanks!

File: src/components/views/settings/SetIdServer.js (hunk 3 above)
Comment (jryans) [reply src/components/views/settings/SetIdServer.js]: Thanks, added.

**Pull request merged**